client/internal/receiver.go

```diff
@@ -168,3 +169,23 @@ func (r *Receiver) processErrorResponse(body *protobufs.ServerErrorResponse) {
 func (r *Receiver) rcvAddonsAvailable(addons *protobufs.AddonsAvailable) {
 	// TODO: implement this.
 }
+
+func (r *Receiver) rcvAgentIdentification(ctx context.Context, agentId *protobufs.AgentIdentification) {
+	if agentId == nil {
+		return
+	}
+
+	err := r.callbacks.OnAgentIdentification(ctx, agentId)
+	if err != nil {
+		r.logger.Errorf("Error while retrieving agent identification: %v", err)
+		return
+	}
+
+	instanceUid := agentId.NewInstanceUid
+	if instanceUid == "" {
+		r.logger.Debugf("Empty instance uid is not allowed")
+		return
+	}
```

Comment:
Should this check happen before we call the callback? Otherwise the callback may update its internal id to empty string but we reject it here.

Reply:
I was thinking mostly about making sure we pass a valid id to `SetInstanceUid` but perhaps it should have its own check and we could move this one up

Reply:
Yes, I agree. I suggest this order:
1. Check non-empty, exit if empty.
2. Callback, exit if error.
3. SetInstanceUid()

Reply:
Just updated the code (including rebase). Now SetInstanceUid might return an error, which is logged here